contracts/MarginCalculator.sol

```diff
@@ -21,18 +20,40 @@ contract MarginCalculator {
     using SafeMath for uint256;
     using FPI for FPI.FixedPointInt;
 
-    address public addressBook;
+    /// @dev oracle module
+    OracleInterface public oracle;
 
     /// @dev decimals used by strike price and oracle price
     uint256 internal constant BASE = 8;
 
     /// @dev FixedPoint 0
     FPI.FixedPointInt internal ZERO = FPI.fromScaledUint(0, BASE);
 
-    constructor(address _addressBook) public {
-        require(_addressBook != address(0), "MarginCalculator: invalid addressbook");
+    struct VaultDetails {
+        address shortUnderlyingAsset;
+        address shortStrikeAsset;
+        address shortCollateralAsset;
+        address longUnderlyingAsset;
+        address longStrikeAsset;
+        address longCollateralAsset;
+        uint256 shortStrikePrice;
+        uint256 shortExpiryTimestamp;
+        uint256 shortCollateralDecimals;
+        uint256 longStrikePrice;
+        uint256 longExpiryTimestamp;
+        uint256 longCollateralDecimals;
+        uint256 collateralDecimals;
+        bool shortType;
+        bool longType;
```

Comment:
I think it's better to call them isShortPut, isLongPut.
IMO it's nontrivial to see a boolean variable that's called "type".